src/MSBuildTaskHost/FileSystem/MSBuildTaskHostFileSystem.cs

```diff
@@ -21,6 +21,26 @@ public bool DirectoryEntryExists(string path)
             return NativeMethodsShared.FileOrDirectoryExists(path);
         }
 
+        public void ClearCaches() { }
```

Comment:
Why is this part of the interface if the impl is empty and it's never called?

Same with the write state method below

Reply:
I also wasn't sure about this one, so left it for review comments :)

Having a method to clear caches would be useful in situations where replacing the entire IFileSsytem object is not possible (the msbuild xml cache suffers from this pain). But this is speculative, since no particular situation comes to mind.

Reply:
I'm fine with leaving it or removing it, but if you leave it, I'd prefer for it to be implemented. I don't know if the intention is for a user to be able to call this via IFileSystem, but if so, they would be confused that it doesn't do anything. (And I would be, too, if I saw the method existed and tried to call it but couldn't figure out what exactly it did until I looked at it.)